src/main/java/com/exactpro/th2/FixHandler.java

```diff
@@ -169,7 +168,7 @@ public class FixHandler implements AutoCloseable, IHandler {
     private final AtomicReference<Future<?>> testRequestTimer = new AtomicReference<>(CompletableFuture.completedFuture(null));
 
     private final SendingTimeoutHandler sendingTimeoutHandler;
-    private Future<?> reconnectRequestTimer = CompletableFuture.completedFuture(null);
+    private AtomicReference<Future<?>> reconnectRequestTimer = new AtomicReference<>(CompletableFuture.completedFuture(null));
```

Comment:
Why have you decided to use AtomicReference here? It looks like we could use volatile here instead since we only use `set` and `get` here